tests/end_to_end/features/fixtures.py

```diff
@@ -0,0 +1,33 @@
+from azure.identity import DefaultAzureCredential
+from azure.storage.filedatalake import DataLakeDirectoryClient
+from constants import ADLS_URL, SQL_DB_INGEST_DIRECTORY_NAME, RAW_CONTAINER_NAME
+from azure.core.exceptions import ResourceNotFoundError
+from behave import fixture
+
+
+def delete_directory_adls(client: DataLakeDirectoryClient, directory_path: str):
```

Comment:
Again nit picking again, but I prefer not to use exceptions as control flow (https://softwareengineering.stackexchange.com/questions/189222/are-exceptions-as-control-flow-considered-a-serious-antipattern-if-so-why). You should have an exception handler here, for errors you can't predict, I tend to check and raise.

But just tips - not essential

Reply:
Will need some further clarification as to what you mean here. This function runs both before and after the test to ensure a clean environment. Typically we would expect the directory to exist after the test has run but not before as it should have been cleaned up. `ResourceNotFoundError` is the error that we get if the directory does not exist therefore it needs to be handled and we want to proceed with the code. The following block
 ```
   except Exception:
        raise
```
will raise any of the other exceptions we may get that can't predict allowing these to be seen and handled in the output log.

Reply:
So I would check if file exists and raise a specific error type, rather than catching and raising. We can chat if we want.

Reply:
Ah I see what you mean. I agree this is a better way to handle it

Reply:
Done here 0eebcb47b748c7cbe3cd4c5a82db50e2b05d2674